src/lib/learn-client/api/progress/get-all-progress.ts

```diff
@@ -0,0 +1,56 @@
+import queryString from 'query-string'
+import { get, toError } from 'lib/learn-client'
+import { Tutorial, Collection } from 'lib/learn-client/types'
+import { SessionData } from 'types/auth'
+import { PROGRESS_API_ROUTE } from '.'
+import { ApiCollectionTutorialProgress } from '../api-types'
+
+interface GetAllProgressOptions {
+	accessToken: SessionData['accessToken']
+	tutorialIds?: Tutorial['id'][]
+	collectionIds?: Collection['id'][]
+}
+
+type GetAllProgressResult = ApiCollectionTutorialProgress[]
+
+/**
+ * Fetches all tutorial progress records for the current user.
+ * If successful, returns an array of progress objects.
+ * Otherwise throws an error.
+ *
+ * Accepts optional arrays of `tutorialIds` and `collectionIds`.
+ * When either of these are provided, we return all progress records
+ * that match either:
+ * - any one of the provided `tutorialIds`
+ * - any one of the provided `collectionIds`
+ *
+ * https://digital-api-specs.vercel.app/learn#tag/Progress/paths/~1progress/get
+ */
+const getAllProgress = async ({
+	accessToken,
+	tutorialIds,
+	collectionIds,
+}: GetAllProgressOptions): Promise<GetAllProgressResult> => {
+	// Add query params to the URL, if applicable
+	let url = PROGRESS_API_ROUTE
+	const qs = queryString.stringify({ tutorialIds, collectionIds })
+	if (qs !== '') {
+		url += `?${qs}`
+	}
+
+	// Make the GET request
+	const requestResult = await get(url, accessToken)
```

Comment:
🙋🏻 Do we need to handle a 404 status?

Reply:
I don't think so - unlike with individual requests, we're always hitting the same route (`/progress`), so a 404 would be very unexpected. Even if a user has no progress records, we expect back `data` that's an empty array.